app/src/main/java/com/infomaniak/mail/data/MailData.kt

```diff
@@ -157,22 +163,90 @@ object MailData {
         }
     }
 
-    fun loadThreads(folder: Folder, mailbox: Mailbox, offset: Int) {
+    fun loadThreads(folder: Folder, mailbox: Mailbox, offset: Int, forceRefresh: Boolean = false) {
+        val isInternetAvailable = true // TODO: Manage this for real
         val realmThreads = getThreadsFromRealm(folder, offset)
-        getThreadsFromApi(folder, mailbox, offset, realmThreads)
+
+        if (Folder.isDraftsFolder() && isInternetAvailable) {
+            val realmOfflineDrafts = realmThreads
+                .flatMap { it.messages }
+                .filter { it.isDraft }
+                .mapNotNull { it.draftUuid?.let(MailboxContentController::getDraft) }
+                .filter { it.isOffline || it.isModifiedOffline }
+
+            CoroutineScope(Dispatchers.IO).launch {
+                realmOfflineDrafts.forEach { draft -> saveOfflineDraftToApi(draft) }
+                getThreadsFromApi(folder, mailbox, realmThreads, offset, forceRefresh)
+            }
+        } else {
+            getThreadsFromApi(folder, mailbox, realmThreads, offset, forceRefresh)
+        }
     }
```

Comment:
Why it's not a viewModel ?

Reply:
It's not in the viewModel yet because the refactor of the architecture will be merged after this PR (and mailData will be entirely suppressed), so it will be put in viewModel in the architecture PR to avoid too many changes that will be deleted afterwards

Reply:
ok